ssl/ech/ech_ssl_apis.c

```diff
@@ -13,70 +13,311 @@
 
 int SSL_CTX_set1_echstore(SSL_CTX *ctx, OSSL_ECHSTORE *es)
 {
-    return 0;
+    if (ctx == NULL || es == NULL) {
```

Comment:
I might have expected it to be valid to have `es` be NULL to clear any previously set value without setting a new one yet.

Reply:
ah, hadn't considered that - I guess it's a generic pattern to have `FOO_set1_bar(NULL)` free up the internal `bar` struct? given that, will make this and related changes (shortly).

Reply:
Yes, this is typical for other similar functions.

Reply:
ack - I'll check the other `*_set1_*()` instances and see if others need similar changes (likely tomorrow+)

Reply:
I found a couple more instances where that semantic needed a change, so the push below from just now fixes those up.